packages/core/src/service/services/channel.service.ts

```diff
@@ -124,6 +119,42 @@ export class ChannelService {
         return entity;
     }
 
+    /**
+     * This method is used to bypass a bug with Typeorm when working with ManyToMany relationships.
+     * For some reason, a regular query does not return all the channels that an entity has.
+     * This is a most optimized way to get all the channels that an entity has.
+     *
+     * @param ctx - The RequestContext object.
+     * @param entityType - The type of the entity.
+     * @param entityId - The ID of the entity.
+     * @returns A promise that resolves to an array of objects, each containing a channel ID.
+     * @private
+     */
+    private async getAssignedEntityChannels<T extends ChannelAware & VendureEntity>(ctx: RequestContext, entityType: Type<T>, entityId: T['id']): Promise<{ channelId: ID }[]> {
+        const repository = this.connection.getRepository(ctx, entityType);
+
+        const metadata = repository.metadata;
+        const channelsRelation = metadata.findRelationWithPropertyPath('channels');
+
+        if (!channelsRelation) {
+            throw new InternalServerError(`Could not find the channels relation for entity ${metadata.name}`);
+        }
+
+        const junctionTableName = channelsRelation.junctionEntityMetadata?.tableName;
+        const junctionColumnName = channelsRelation.junctionEntityMetadata?.columns[0].databaseName;
+        const inverseJunctionColumnName = channelsRelation.junctionEntityMetadata?.inverseColumns[0].databaseName;
```

Comment:
With the latest updates we explicitly added inverse relations to the Channel entity. 

I'm wondering what would happen if someone defines a custom channel-aware entity which does not have the inverse relation on the Channel? Would this break?

Reply:
This is quite normal, there should be no errors because even without reverse entity feedback is still available
But if there are problems you we replace it with pure SQL

> With the latest updates we explicitly added inverse relations to the Channel entity.
> 
> I'm wondering what would happen if someone defines a custom channel-aware entity which does not have the inverse relation on the Channel? Would this break?

Reply:
or just manually parse the column names in the table. This can also be done. We just need to test it on a real project